scripts/docker-as-symfony28

```diff
@@ -0,0 +1,456 @@
+#!/usr/bin/env bash
+set -e
+
+# Environement variables
+: ${DATABASE_URL:="mysql://root:root@db/drupal"}
+: ${DATABASE_DUMP:="/var/backups/db-reset.sql"}
+: ${SMTP_HOST:="mail:1025"}
+: ${REQUIRED_DIRECTORIES:=""}
+: ${LOCK_TIME:=""}
+: ${TEST_SERVER_PORT:="8888"}
+
+# Script variables
+DATABASE_HOST=$(a=${DATABASE_URL#*@}; echo ${a%/*})
+DATABASE_NAME=${DATABASE_URL##*/}
+DATABASE_USER=$(a=${DATABASE_URL#*//}; echo ${a%:*})
+DATABASE_PASSWORD=$(a=${DATABASE_URL#*//}; b=${a#*:}; echo ${b%@*})
+SYMFONY_IS_INSTALLED=0
+SCRIPT_NAME=$(basename $0)
+
+# Print help
+function print_help() {
+  echo "usage: $SCRIPT_NAME <action> [options]
+
+  * bootstrap
+
+    Install Symfony and dependencies. Require config export, do not bootstrap a new projet.
+
+    --skip-dependencies      # Do not run composer and yarn install
+    --skip-install           # Do not run Drupal install (only if arealdy installed)
+    --skip-db-reset          # Do not reset database using SQL dump (only when --skip-install)
+    --install-only           # Same as --skip-dependencies --skip-styleguide-build
+
+  * setup
+
+    Ensure Symfony is porper setup (parameters.yml, mail settings, ...).
+
+    --with-dependencies      # Run composer and yarn install
+
+  * db-reset
+
+    Reset database using SQL dump made after bootstrap.
+
+    --update-dump            # Update database dump (include updated Drupal config, but not
+                             # default content)
+
+  * db-dump
+
+    Create a database dump that can be used as with db-restore (not with db-reset).
+
+    --file=<path>            # Path to dump file (not required)
+
+  * db-restore
+
+    Restore database from dump created with db-dump.
+
+    --file=<path>            # Path to dump file (not required)
+
+  * apache-server
+
+    Run web server using 'apache'. Reset database if not present.
+
+    --with-db-reset          # Reset database before launch server
+    --cache                  # Do not invalidate opcache (server must be restarted
+                             # reload cache)
+    --help                   # Display apache help
+    -- <...>                 # any apache valid args
+  "
+}
+
+# Setup
+function setup() {
+  printf "\e[1;35m* Update parameters.yml file.\e[0m\n"
+
+  if [ ! -f /var/www/app/config/parameters.yml ]; then
+    cp /var/www/app/config/parameters.yml.dist /var/www/app/config/parameters.yml
+  fi
+
+  # Populate database config
+  sed -ri -e "s@database_host:.*@database_host: ${DATABASE_HOST}@g" /var/www/app/config/parameters.yml
+  sed -ri -e "s@database_name:.*@database_name: ${DATABASE_NAME}@g" /var/www/app/config/parameters.yml
+  sed -ri -e "s@database_user:.*@database_user: ${DATABASE_USER}@g" /var/www/app/config/parameters.yml
+  sed -ri -e "s@database_password:.*@database_password: ${DATABASE_PASSWORD}@g" /var/www/app/config/parameters.yml
+
+  # Populate mailer config
+  sed -ri -e "s@mailer_host:.*@mailer_host: ${SMTP_HOST%:*}@g" /var/www/app/config/parameters.yml
```

Comment:
I really don't like this!
I recommend a way to setup this with ENV variables (there is a way in Symfony 2.8 too https://symfony.com/doc/2.8/configuration/external_parameters.html )

Reply:
This script is for a legacy symfony version and is currently used in one project: https://github.com/antistatique/rmi-platform/blob/dev/codeship-steps.yml

I'm for moving this script out of this repo and include it only the specific project.

Reply:
We could, it's legacy yes and not plan to maintain it.

Reply:
done: https://github.com/antistatique/rmi-platform/commit/6c75111aea476f1bb5f8a568e8b368d186d2798f